### PR TITLE
Add Tracks Events to Google My Business Stats Nudge

### PR DESCRIPTION
**Pull request by cbauerman**
This PR adds Tracks Events to the new Google My Business Stats Nudge (#22221) 

## Specs

The events to be added are labeled on the below image:

m1-nudge1-events

1. `calypso_google_my_business_stats_nudge_shown` whenever the nudge is shown
2. `calypso_google_my_business_stats_nudge_dismissed` whenever the nudge is dismissed via the x in the corner
3. `calypso_google_my_business_stats_nudge_start_now_clicked` whenever the user clicks the "Start Now" button to proceed to the next part of the flow.
4. `calypso_google_my_business_stats_nudge_already_listed_clicked` whenever the user clicks the "I've already listed" button.

The events will have no special properties initially.

## Testing

- Boot this branch locally or use the calypso.live link below
- Enable debug of tracks in Calypso by typing this in your browser console:  `localStorage.setItem('debug', 'calypso:analytics:tracks' )`
- Create a site with a `promote` siteGoals if you don't have one. This can be done from the `http://calypso.localhost:3000/start/about` page by selecting at least the "Promote your business, skills, organization, or events" option.
- Upgrade your site plan to business or premium
- Navigate to this site stats page `/stats/day/:site_slug`
- Notice the Google My Business nudge
- Notice in the console if a `calypso_test_google_my_business_stats_nudge_shown` event was recorded
- Press the two buttons on the nudge and dismiss it and notice in the console if the correct events were recorded for each.

`localStorage.removeItem(  'debug' )` will stop the event logging

## Before Merging
- [ ] Remove `test_` from each event name

### CODE REVIEW

File: client/components/google-my-business/stats-nudge.js [hunk 1]
```diff
@@ -3,52 +3,78 @@
...
-							'Show up when customers search for businesses like yours on Google Search and Maps.'
-						) }
-					</h2>
-					<div className="google-my-business__stats-nudge-button-row">
-						<Button primary>{ translate( 'Start Now' ) }</Button>
-						<Button>{ translate( "I've Already Listed" ) }</Button>
+class GoogleMyBusinessStatsNudge extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+	};
```
Comment (Tug) [inline client/components/google-my-business/stats-nudge.js]: We can actually add the props functions declared in `connect` here.

File: client/components/google-my-business/stats-nudge.js [hunk 2]
```diff
@@ -3,52 +3,78 @@
...
 				</div>
-			</div>
-		</DismissibleCard>
-	);
-};
+			</DismissibleCard>
+		);
+	}
+
+	componentWillMount() {
```
Comment (Tug) [inline client/components/google-my-business/stats-nudge.js]: Minor but we usually have `componentWillMount` at the beginning.

**Comment by cbauerman** [comment]
Yeah, we will probably be clicking those buttons for a while in development, so keeping the `test` in there will be helpful for a while.

File: client/components/google-my-business/stats-nudge.js [hunk 3]
```diff
@@ -3,52 +3,82 @@
...
+			</DismissibleCard>
+		);
+	}
+}
 
-export default localize( GMBStatsNudge );
+export default connect( () => ( {} ), {
+	trackNudgeShown: () => recordTracksEvent( 'calypso_test_google_my_business_stats_nudge_shown' ),
+	trackNudgeDismissed: () =>
+		recordTracksEvent( 'calypso_test_google_my_business_stats_nudge_dismissed' ),
```
Comment (stephanethomas) [inline client/components/google-my-business/stats-nudge.js]: Shouldn't this function and Track event have a `clicked` suffix like other other actions?

**Pull request merged**